run_fastsurfer.sh

```diff
@@ -504,6 +504,49 @@ esac
 done
 set -- "${POSITIONAL[@]}" # restore positional parameters
 
+
+# Function to convert path to absolute if it exists and is not empty
+convert_to_absolute_path() {
+  local var_name="$1"
+  local var_value
+  eval "var_value=\$$var_name"
+  
+  if [ -n "$var_value" ]; then
+    if [ -e "$var_value" ] || [ "${var_value#*/}" != "$var_value" ]; then
+      # Path exists or contains directory separators - convert to absolute
+      local abs_path
+      abs_path=$(realpath "$var_value" 2>/dev/null)
+      if [ $? -eq 0 ] && [ -n "$abs_path" ]; then
+        eval "$var_name=\"$abs_path\""
+      else
+        echo "WARNING: Failed to convert $var_name to absolute path: $var_value" | tee -a "${tmpLF:-/dev/null}"
```

Comment:
how you can achieve the `var=$(func "$val")"` syntax is by echo-ing the return value and sending warnings and error messages to stderr.
```suggestion
  # usage: convert_to_absolute_path <path> [<logfile>]
  local var_value="$1"
  local logging=("cat")
  if [[ "$#" > 1]] ; then logging=(tee -a "$2") ; fi
  
  if [ -n "$var_value" ]; then
    if [ -e "$var_value" ] || [ "${var_value#*/}" != "$var_value" ]; then
      # Path exists or contains directory separators - convert to absolute
      local abs_path
      abs_path=$(realpath "$var_value" 2>/dev/null)
      if [ $? -eq 0 ] && [ -n "$abs_path" ]; then
        echo "$abs_path"
      else
        echo "WARNING: Failed to convert $var_name to absolute path: $var_value" | "${logging[@]}" &>2
```

Overall, this function should probably be moved to `functions.sh`.

Also, logging can be done more elegantly like indicated.